### ISSUE TITLE
Bug in archive_apk_packages

### ISSUE THREAD

**Issue by jowiho** [elastic/gradle-plugins#37]
The recently introduced `archive_apk_packages` function in `archive-packages.sh` does not properly split package names and version when the package name contains a hyphen (`-`).

The offending code assumes that everything before the first hyphen is the package name and everything after is the package version, which of course isn't correct when the package name itself contains a hyphen, like `apk-tools`.
https://github.com/elastic/gradle-plugins/blob/2ad060e6b17909e3c73249d6491134fac487eb3d/plugins/docker/base-image/src/main/resources/archive-packages.sh#L103-L104

As a result, the following `apk add $package_version_list` step fails with
```
ERROR: 'apk=tools-2.14.4-r0' is not a valid world dependency, format is name(@tag)([<>~=]version)
```
I'm not sure what the consequences of that failure are exactly.

**Comment by jowiho** [comment]
Instead of doing `apk info` followed by ` apk fetch --url --simulate`, you could also get the list of packages, versions and architectures (which BTW are currently always blank in the lock file) from `apk list --installed`:
```
apk-tools-2.14.4-r0 x86_64 {apk-tools} (GPL-2.0-only) [installed]
busybox-1.36.1-r10 x86_64 {busybox} (GPL-2.0-only) [installed]
...
```

**Comment by imkarrer** [comment]
I think we leverage the URLs to make sure we get the latest packages.

https://github.com/elastic/gradle-plugins/blob/b8b82369191ad03724d00c0e3cde4bc443b53b11/plugins/docker/base-image/src/main/resources/archive-packages.sh#L99

I suppose I can do both and use the URLs to get the latest packages and the `apk list` to get the arch`.  Which I am not sure we need to do since we can just use PACKAGES_ARCH, https://github.com/elastic/gradle-plugins/blob/d08792376808cc523dea3aadd7aa53023a772019/plugins/docker/base-image/src/main/resources/archive-packages.sh#L125

**Comment by alpar-t** [comment]
I'm not sure the lack of architecture in the lock-file is an actual issue, wolfi packages don't have the architecture in the package name. That field is meant to track the package name, e.g. ubuntu and/or centos do have `all` or an architecture in the package name.

**Comment by andreaturli** [comment]
FYI this is blocking https://github.com/elastic/cloud/pull/129862 as when I run `:found-deployment:docker:billing-service:dockerBaseImageLockfile` it fails with https://gradle-enterprise.elastic.co/s/gotkn7jhz5x34

**Comment by imkarrer** [comment]
@andreaturli `This build did not contain any failures.`
Got another link or a stacktrace? I am actively working in this area.

**Comment by andreaturli** [comment]
let me add the output I can see locally

```
17:14:17: Executing 'dockerBaseImageLockfile'...

> Task :buildSrc:compileKotlin UP-TO-DATE
> Task :buildSrc:compileJava UP-TO-DATE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:pluginDescriptors UP-TO-DATE
> Task :buildSrc:processResources UP-TO-DATE
> Task :buildSrc:classes UP-TO-DATE
> Task :buildSrc:jar UP-TO-DATE
> Task :buildSrc:assemble UP-TO-DATE
> Task :buildSrc:compileTestKotlin NO-SOURCE
> Task :buildSrc:compileTestJava UP-TO-DATE
> Task :buildSrc:compileTestGroovy NO-SOURCE
> Task :buildSrc:pluginUnderTestMetadata UP-TO-DATE
> Task :buildSrc:processTestResources UP-TO-DATE
> Task :buildSrc:testClasses UP-TO-DATE
> Task :buildSrc:test UP-TO-DATE
> Task :buildSrc:validatePlugins UP-TO-DATE
> Task :buildSrc:check UP-TO-DATE
> Task :buildSrc:build UP-TO-DATE
=======================================
Elastic Cloud Build Gerbil says Hello!
  Gradle Version        : 7.6.4
  Gradle home           : /Users/andrea/.gradle/wrapper/dists/gradle-7.6.4-all/69w292fwwbbevpookp42lrbux/gradle-7.6.4
  Gradle user home      : /Users/andrea/.gradle
  OS Info               : Mac OS X 14.5 (aarch64)
  JDK Version           : 17
  JAVA_HOME             : /Users/andrea/Library/Java/JavaVirtualMachines/temurin-17.0.11/Contents/Home
=======================================
> Task :syncBinDirStaticCli UP-TO-DATE

> Task :found-deployment:docker:billing-service:dockerBaseImageLockfile
#0 building with "default" instance using docker driver

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 893B done
#1 DONE 0.0s

#2 [internal] load metadata for docker.elastic.co/wolfi/chainguard-base:20230214@sha256:a9a7ca1e7fd7719480da391e8293ecfb5f55f2735b1b8e946419f57670e48017
#2 DONE 0.0s

#3 [internal] load .dockerignore
#3 transferring context: 69B done
#3 DONE 0.0s

#4 [stage-0 1/3] FROM docker.elastic.co/wolfi/chainguard-base:20230214@sha256:a9a7ca1e7fd7719480da391e8293ecfb5f55f2735b1b8e946419f57670e48017
#4 CACHED

#5 [internal] load build context
#5 transferring context: 67B done
#5 DONE 0.0s

#6 [stage-0 2/3] RUN --mount=type=bind,readonly,target=/mnt/ephemeral,source=ephemeral/docker apk add python-3.11=3.11.9-r5 py3-pip=24.1.2-r0 runit=2.1.2-r2
#6 0.116 fetch https://packages.cgr.dev/extras/aarch64/APKINDEX.tar.gz
#6 0.550 fetch https://packages.wolfi.dev/os/aarch64/APKINDEX.tar.gz
#6 1.915 (1/16) Installing py3-pip (24.1.2-r0)
#6 2.053 (2/16) Installing libbz2-1 (1.0.8-r8)
#6 2.211 (3/16) Installing libexpat1 (2.6.2-r1)
#6 2.558 (4/16) Installing libffi (3.4.6-r3)
#6 2.720 (5/16) Installing gdbm (1.24-r0)
#6 3.094 (6/16) Installing xz (5.6.2-r0)
#6 3.359 (7/16) Installing libgcc (13.3.0-r1)
#6 3.520 (8/16) Installing libstdc++ (13.3.0-r1)
#6 4.012 (9/16) Installing mpdecimal (4.0.0-r2)
#6 4.226 (10/16) Installing ncurses-terminfo-base (6.5_p20240629-r0)
#6 4.607 (11/16) Installing ncurses (6.5_p20240629-r0)
#6 4.850 (12/16) Installing readline (8.2-r7)
#6 5.255 (13/16) Installing sqlite-libs (3.45.1-r1)
#6 5.511 (14/16) Installing python-3.11-base (3.11.9-r5)
#6 6.933 (15/16) Installing python-3.11 (3.11.9-r5)
#6 7.296 (16/16) Installing runit (2.1.2-r2)
#6 7.696 OK: 69 MiB in 31 packages
#6 DONE 8.2s

#7 [stage-0 3/3] RUN --mount=type=bind,readonly,target=/mnt/ephemeral,source=ephemeral/docker apk upgrade
#7 0.312 OK: 69 MiB in 31 packages
#7 DONE 0.4s

#8 exporting to image
#8 exporting layers
#8 exporting layers 0.2s done
#8 writing image sha256:f1013425c0070d6ba14e4c8f0581fc2b844e7f0affed446c70d19009e00cf281
#8 writing image sha256:f1013425c0070d6ba14e4c8f0581fc2b844e7f0affed446c70d19009e00cf281 done
#8 naming to docker.io/library/f1de644b-001b-4d7c-a44e-3680ea03f07d done
#8 DONE 0.2s

Running the created image to extract package information and upload packages with /Users/andrea/GradleProjects/cloud/.gradle/bin/jfrog-cli-linux-aarch64 ...
fetch https://packages.cgr.dev/extras/aarch64/APKINDEX.tar.gz
fetch https://packages.wolfi.dev/os/aarch64/APKINDEX.tar.gz
 [https://packages.cgr.dev/extras]
 [https://packages.wolfi.dev/os]
OK: 72182 distinct packages available
Installed packages: apk-tools
busybox
ca-certificates-bundle
gdbm
glibc
glibc-locale-posix
ld-linux
libbz2-1
libcrypt1
libcrypto3
libexpat1
libffi
libgcc
libssl3
libstdc++
libxcrypt
mpdecimal
ncurses
ncurses-terminfo-base
py3-pip
python-3.11
python-3.11-base
readline
runit
sqlite-libs
wolfi-base
wolfi-baselayout
wolfi-keys
xz
zlib
ERROR: 'apk=tools-2.14.4-r0' is not a valid world dependency, format is name(@tag)([<>~=]version)
Adding curl so we can use it to download packages
(1/8) Installing libbrotlicommon1 (1.1.0-r3)
(2/8) Installing libbrotlidec1 (1.1.0-r3)
(3/8) Installing libnghttp2-14 (1.62.1-r1)
(4/8) Installing libunistring (1.2-r2)
(5/8) Installing libidn2 (2.3.7-r2)
(6/8) Installing libpsl (0.21.5-r3)
(7/8) Installing libcurl-openssl4 (8.8.0-r0)
(8/8) Installing curl (8.8.0-r0)
OK: 74 MiB in 39 packages
Downloading: https://packages.wolfi.dev/os/aarch64/apk-tools-2.14.4-r0.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  179k  100  179k    0     0  1130k      0 --:--:-- --:--:-- --:--:-- 1136k
Downloading: https://packages.wolfi.dev/os/aarch64/busybox-1.36.1-r10.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  361k  100  361k    0     0  1003k      0 --:--:-- --:--:-- --:--:-- 1001k
Downloading: https://packages.wolfi.dev/os/aarch64/ca-certificates-bundle-20240705-r0.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  127k  100  127k    0     0   346k      0 --:--:-- --:--:-- --:--:--  347k
Downloading: https://packages.wolfi.dev/os/aarch64/gdbm-1.24-r0.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  319k  100  319k    0     0  1923k      0 --:--:-- --:--:-- --:--:-- 1926k
Downloading: https://packages.wolfi.dev/os/aarch64/glibc-2.39-r7.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1881k  100 1881k    0     0  3297k      0 --:--:-- --:--:-- --:--:-- 3295k
Downloading: https://packages.wolfi.dev/os/aarch64/glibc-locale-posix-2.39-r7.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 75732  100 75732    0     0   480k      0 --:--:-- --:--:-- --:--:--  483k
Downloading: https://packages.wolfi.dev/os/aarch64/ld-linux-2.39-r7.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  103k  100  103k    0     0   299k      0 --:--:-- --:--:-- --:--:--  300k
Downloading: https://packages.wolfi.dev/os/aarch64/libbz2-1-1.0.8-r8.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 40148  100 40148    0     0   333k      0 --:--:-- --:--:-- --:--:--  332k
Downloading: https://packages.wolfi.dev/os/aarch64/libcrypt1-2.39-r7.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2150  100  2150    0     0   7825      0 --:--:-- --:--:-- --:--:--  7846
Downloading: https://packages.wolfi.dev/os/aarch64/libcrypto3-3.3.1-r4.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2070k  100 2070k    0     0  4260k      0 --:--:-- --:--:-- --:--:-- 4268k
Downloading: https://packages.wolfi.dev/os/aarch64/libexpat1-2.6.2-r1.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 75191  100 75191    0     0   465k      0 --:--:-- --:--:-- --:--:--  467k
Downloading: https://packages.wolfi.dev/os/aarch64/libffi-3.4.6-r3.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 23373  100 23373    0     0   195k      0 --:--:-- --:--:-- --:--:--  195k
Downloading: https://packages.wolfi.dev/os/aarch64/libgcc-13.3.0-r1.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 60110  100 60110    0     0   354k      0 --:--:-- --:--:-- --:--:--  355k
Downloading: https://packages.wolfi.dev/os/aarch64/libssl3-3.3.1-r4.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  472k  100  472k    0     0  1312k      0 --:--:-- --:--:-- --:--:-- 1312k
Downloading: https://packages.wolfi.dev/os/aarch64/libstdc++-13.3.0-r1.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1028k  100 1028k    0     0  2078k      0 --:--:-- --:--:-- --:--:-- 2077k
Downloading: https://packages.wolfi.dev/os/aarch64/libxcrypt-4.4.36-r7.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 91603  100 91603    0     0   271k      0 --:--:-- --:--:-- --:--:--  271k
Downloading: https://packages.wolfi.dev/os/aarch64/mpdecimal-4.0.0-r2.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  116k  100  116k    0     0   323k      0 --:--:-- --:--:-- --:--:--  323k
Downloading: https://packages.wolfi.dev/os/aarch64/ncurses-6.5_p20240629-r0.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  424k  100  424k    0     0  2278k      0 --:--:-- --:--:-- --:--:-- 2271k
Downloading: https://packages.wolfi.dev/os/aarch64/ncurses-terminfo-base-6.5_p20240629-r0.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 98942  100 98942    0     0   294k      0 --:--:-- --:--:-- --:--:--  295k
Downloading: https://packages.wolfi.dev/os/aarch64/py3-pip-24.1.2-r0.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2030  100  2030    0     0  15041      0 --:--:-- --:--:-- --:--:-- 15149
Downloading: https://packages.wolfi.dev/os/aarch64/python-3.11-3.11.9-r6.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2342  100  2342    0     0   8195      0 --:--:-- --:--:-- --:--:--  8217
Downloading: https://packages.wolfi.dev/os/aarch64/python-3.11-base-3.11.9-r6.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 15.4M  100 15.4M    0     0  12.7M      0  0:00:01  0:00:01 --:--:-- 12.7M
Downloading: https://packages.wolfi.dev/os/aarch64/readline-8.2-r7.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  358k  100  358k    0     0  1831k      0 --:--:-- --:--:-- --:--:-- 1840k
Downloading: https://packages.wolfi.dev/os/aarch64/runit-2.1.2-r2.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  601k  100  601k    0     0  2977k      0 --:--:-- --:--:-- --:--:-- 2979k
Downloading: https://packages.wolfi.dev/os/aarch64/sqlite-libs-3.45.1-r1.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  811k  100  811k    0     0  3656k      0 --:--:-- --:--:-- --:--:-- 3672k
Downloading: https://packages.wolfi.dev/os/aarch64/wolfi-base-1-r6.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1979  100  1979    0     0   7366      0 --:--:-- --:--:-- --:--:--  7384
Downloading: https://packages.wolfi.dev/os/aarch64/wolfi-baselayout-20230201-r13.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 12154  100 12154    0     0   112k      0 --:--:-- --:--:-- --:--:--  111k
Downloading: https://packages.wolfi.dev/os/aarch64/wolfi-keys-1-r8.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2984  100  2984    0     0  11261      0 --:--:-- --:--:-- --:--:-- 11303
Downloading: https://packages.wolfi.dev/os/aarch64/xz-5.6.2-r0.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  425k  100  425k    0     0  1943k      0 --:--:-- --:--:-- --:--:-- 1951k
Downloading: https://packages.wolfi.dev/os/aarch64/zlib-1.3.1-r3.apk
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 52088  100 52088    0     0   188k      0 --:--:-- --:--:-- --:--:--  188k
Removing curl so it's not part of the image
(1/8) Purging curl (8.8.0-r0)
(2/8) Purging libcurl-openssl4 (8.8.0-r0)
(3/8) Purging libnghttp2-14 (1.62.1-r1)
(4/8) Purging libpsl (0.21.5-r3)
(5/8) Purging libbrotlidec1 (1.1.0-r3)
(6/8) Purging libbrotlicommon1 (1.1.0-r3)
(7/8) Purging libidn2 (2.3.7-r2)
(8/8) Purging libunistring (1.2-r2)
OK: 69 MiB in 31 packages
All files have been downloaded to /var/cache/apk/archives/aarch64
Recording: apk-tools=2.14.4-r0 in the lockfile
Recording: busybox=1.36.1-r10 in the lockfile
Recording: ca-certificates-bundle=20240705-r0 in the lockfile
Recording: gdbm=1.24-r0 in the lockfile
Recording: glibc=2.39-r7 in the lockfile
Recording: glibc-locale-posix=2.39-r7 in the lockfile
Recording: ld-linux=2.39-r7 in the lockfile
Recording: libbz2-1=1.0.8-r8 in the lockfile
Recording: libcrypt1=2.39-r7 in the lockfile
Recording: libcrypto3=3.3.1-r4 in the lockfile
Recording: libexpat1=2.6.2-r1 in the lockfile
Recording: libffi=3.4.6-r3 in the lockfile
Recording: libgcc=13.3.0-r1 in the lockfile
Recording: libssl3=3.3.1-r4 in the lockfile
Recording: libstdc++=13.3.0-r1 in the lockfile
Recording: libxcrypt=4.4.36-r7 in the lockfile
Recording: mpdecimal=4.0.0-r2 in the lockfile
Recording: ncurses=6.5_p20240629-r0 in the lockfile
Recording: ncurses-terminfo-base=6.5_p20240629-r0 in the lockfile
Recording: py3-pip=24.1.2-r0 in the lockfile
Recording: python-3.11=3.11.9-r5 in the lockfile
Recording: python-3.11-base=3.11.9-r5 in the lockfile
Recording: readline=8.2-r7 in the lockfile
Recording: runit=2.1.2-r2 in the lockfile
Recording: sqlite-libs=3.45.1-r1 in the lockfile
Recording: wolfi-base=1-r6 in the lockfile
Recording: wolfi-baselayout=20230201-r13 in the lockfile
Recording: wolfi-keys=1-r8 in the lockfile
Recording: xz=5.6.2-r0 in the lockfile
Recording: zlib=1.3.1-r3 in the lockfile
Index has 0 packages (of which 30 are new)
15:14:48 [Info] [Thread 8] Uploading artifact: aarch64/python-3.11-3.11.9-r6.apk
15:14:48 [Info] [Thread 6] Uploading artifact: aarch64/wolfi-base-1-r6.apk
15:14:48 [Info] [Thread 3] Uploading artifact: aarch64/xz-5.6.2-r0.apk
15:14:48 [Info] [Thread 5] Uploading artifact: aarch64/runit-2.1.2-r2.apk
15:14:48 [Info] [Thread 1] Uploading artifact: aarch64/wolfi-keys-1-r8.apk
15:14:48 [Info] [Thread 9] Uploading artifact: aarch64/zlib-1.3.1-r3.apk
15:14:48 [Info] [Thread 2] Uploading artifact: aarch64/readline-8.2-r7.apk
15:14:48 [Info] [Thread 0] Uploading artifact: aarch64/wolfi-baselayout-20230201-r13.apk
15:14:48 [Info] [Thread 4] Uploading artifact: aarch64/sqlite-libs-3.45.1-r1.apk
15:14:48 [Info] [Thread 7] Uploading artifact: aarch64/python-3.11-base-3.11.9-r6.apk
15:14:49 [Info] [Thread 4] Uploading artifact: aarch64/py3-pip-24.1.2-r0.apk
15:14:49 [Info] [Thread 7] Uploading artifact: aarch64/ncurses-terminfo-base-6.5_p20240629-r0.apk
15:14:49 [Info] [Thread 3] Uploading artifact: aarch64/ncurses-6.5_p20240629-r0.apk
15:14:49 [Info] [Thread 2] Uploading artifact: aarch64/libxcrypt-4.4.36-r7.apk
15:14:49 [Info] [Thread 9] Uploading artifact: aarch64/mpdecimal-4.0.0-r2.apk
15:14:49 [Info] [Thread 0] Uploading artifact: aarch64/libstdc++-13.3.0-r1.apk
15:14:49 [Info] [Thread 5] Uploading artifact: aarch64/libssl3-3.3.1-r4.apk
15:14:49 [Info] [Thread 6] Uploading artifact: aarch64/libgcc-13.3.0-r1.apk
15:14:49 [Info] [Thread 8] Uploading artifact: aarch64/libffi-3.4.6-r3.apk
15:14:49 [Info] [Thread 1] Uploading artifact: aarch64/libexpat1-2.6.2-r1.apk
15:14:49 [Info] [Thread 7] Uploading artifact: aarch64/libcrypto3-3.3.1-r4.apk
15:14:49 [Info] [Thread 9] Uploading artifact: aarch64/libcrypt1-2.39-r7.apk
15:14:49 [Info] [Thread 3] Uploading artifact: aarch64/libbz2-1-1.0.8-r8.apk
15:14:49 [Info] [Thread 5] Uploading artifact: aarch64/ld-linux-2.39-r7.apk
15:14:49 [Info] [Thread 2] Uploading artifact: aarch64/glibc-locale-posix-2.39-r7.apk
15:14:49 [Info] [Thread 0] Uploading artifact: aarch64/glibc-2.39-r7.apk
15:14:50 [Info] [Thread 4] Uploading artifact: aarch64/gdbm-1.24-r0.apk
15:14:50 [Info] [Thread 6] Uploading artifact: aarch64/ca-certificates-bundle-20240705-r0.apk
15:14:50 [Info] [Thread 1] Uploading artifact: aarch64/busybox-1.36.1-r10.apk
15:14:50 [Info] [Thread 8] Uploading artifact: aarch64/apk-tools-2.14.4-r0.apk
15:14:50 [Info] [Thread 7] Uploading artifact: aarch64/__META__Packages-59224bd6e69c81a313a751fb1007e8bc1fb7b930e1c50513dfb5e64083049af7.gz
{
  "status": "success",
  "totals": {
    "success": 31,
    "failure": 0
  }
}
Written new lockfile to /Users/andrea/GradleProjects/cloud/found-deployment/docker/billing-service/docker-base-image.lock
Untagged: f1de644b-001b-4d7c-a44e-3680ea03f07d:latest
Deleted: sha256:f1013425c0070d6ba14e4c8f0581fc2b844e7f0affed446c70d19009e00cf281

BUILD SUCCESSFUL in 35s
12 actionable tasks: 1 executed, 11 up-to-date

Publishing build scan...
https://gradle-enterprise.elastic.co/s/havhnmmqhnnau

17:14:52: Execution finished 'dockerBaseImageLockfile'.
```

see `ERROR: 'apk=tools-2.14.4-r0' is not a valid world dependency, format is name(@tag)([<>~=]version)`

**Comment by andreaturli** [comment]
> > see ERROR: 'apk=tools-2.14.4-r0' is not a valid world dependency, format is name(@tag)([<>~=]version)
> 
> Despite this error, the plugins still seem to generate a valid lock file (in my case, but apparently also in your case).

I should simply ignore that error, and move on?

**Comment by jowiho** [comment]
> I should simply ignore that error, and move on?

That's what I have done so far. Though you'll likely run into [https://elasticco.atlassian.net/browse/ENGPRD-668](https://elasticco.atlassian.net/browse/ENGPRD-668) next, which you cannot ignore unfortunately.

**Comment by andreaturli** [comment]
oh ok, thanks!

I tried to run manually `bin/gradle_verification_metadata.sh` but the `❯ ./gradlew found-deployment:docker:billing-service:dockerBaseImageLockfileAllWithEmulation` fails with https://gradle-enterprise.elastic.co/s/a65edrioxesdm
`ERROR: failed to solve: process "/bin/sh -c apk add python-3.11=3.11.9-r6 py3-pip=24.1.2-r0 runit=2.1.2-r2" did not complete successfully: exit code: 3`

while I can manually run `add python-3.11=3.11.9-r6 py3-pip=24.1.2-r0 runit=2.1.2-r2` command in a docker container

**Comment by jowiho** [comment]
The gradle plugins do not support version pinning in the `install( )` section. Is there any reason that you want to pin those versions there instead of in the lockfile?

BTW, I've opened a feature request some time ago to discuss if we do need an official way to pin versions: https://github.com/elastic/gradle-plugins/issues/39

**Comment by andreaturli** [comment]
if the lock file is specifying a version, I don't think I need to specify a particular version of, say, `python-3.11`

**Comment by andreaturli** [comment]
but it feels the problem is still there if I don't pin the version
```
 > [stage-0 2/3] RUN --mount=type=bind,readonly,target=/mnt/ephemeral,source=ephemeral/docker apk add python-3.11 py3-pip runit:
0.451 fetch https://packages.wolfi.dev/os/x86_64/APKINDEX.tar.gz
0.451 WARNING: opening from cache https://packages.cgr.dev/extras: No such file or directory
0.730 WARNING: opening from cache https://packages.wolfi.dev/os: No such file or directory
0.734 ERROR: unable to select packages:
0.735   py3-pip (no such package):
0.735     required by: world[py3-pip]
0.735   python-3.11 (no such package):
0.735     required by: world[python-3.11]
0.735   runit (no such package):
0.735     required by: world[runit]
------
```

**Comment by imkarrer** [comment]
I think I've got something that will satisfy reviewers in https://github.com/elastic/gradle-plugins/pull/40, just waiting on the build and ✅

**Comment by andreaturli** [comment]
thanks @imkarrer ! once merged, do we need to release a new version of the gradle plugins?

**Comment by imkarrer** [comment]
Yes, I will publish and raise a PR on cloud

**Comment by imkarrer** [comment]
published.
